types/types.d.ts

```diff
@@ -48,13 +48,13 @@ declare namespace FastifySessionPlugin {
     set(
       sessionId: string,
       session: Fastify.Session,
-      callback: (err: Error) => void
+      callback: (err?: Error) => void
     ): void;
     get(
       sessionId: string,
-      callback: (err: Error, session: Fastify.Session) => void
+      callback: (err: Error | null, session: Fastify.Session) => void
```

Comment:
I tried to use overloading so you either get an `err` _or_ a `session`, but I cannot make typescript understand that for whatever reason.

This is the way the core node types are typed, so probably fine, e.g.: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0beca137d8552f645064b8a622a6e153864c66ee/types/node/fs.d.ts#L898

Reply:
IMO that's the correct way to do it if we refer to the [typescript _Do's and Don'ts_ handbook section](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#overloads-and-callbacks).

Reply:
The type of `callback` here should be `(err?: Error, session?: Fastify.Session) => void`. When retrieving a session, let's say from a database, you can get either `session` or `error`, but never both.

But the type `(err: Error | null, session: Fastify.Session) => void` makes session required. So the callback accept either session, or session + error.

Reply:
again, that's not how node core types work

Reply:
Since callback always check with this pattern.
```js
function callback(err, session) {
  if(err) {
    // do something
    // ...
    return
  }
  
  // do somthing
  session 
}
```
If the session is `?: Fastify.Session` in the above case, the user will be inconvenient and need to double check the existence of `session`.

I would accept the current syntax.